src/verifiable-presentation.js

```diff
@@ -39,6 +39,42 @@ class VerifiablePresentation {
     this.proof = null;
   }
 
+  static fromJSON(json) {
+    const {
+      verifiableCredential, id, type, ...rest
+    } = json;
+    const vp = new VerifiablePresentation(id);
+
+    if (verifiableCredential) {
+      if (verifiableCredential.length) {
+        verifiableCredential.forEach((credential) => {
+          vp.addCredential(credential);
+        });
+      } else {
+        vp.addCredential(verifiableCredential);
+      }
+    }
+
+    if (type) {
+      if (type.length) {
+        type.forEach((typeVal) => {
+          vp.addType(typeVal);
+        });
+      } else {
+        vp.addType(type);
+      }
+    }
+
+    const context = rest['@context'];
+    if (context) {
+      vp.context = rest['@context'];
+      delete rest['@context'];
+    }
+
```

Comment:
Not having a context should raise an error as it won't be a valid JSON-LD object.

Reply:
good point, resolved